amundsen_application/static/js/components/SearchPage/index.tsx

```diff
@@ -117,11 +126,12 @@ export class SearchPage extends React.Component<SearchPageProps> {
   }
 
   shouldUpdateSearchTerm(nextUrlParams, prevUrlParams): boolean {
-    return nextUrlParams.term !== prevUrlParams.term;
+    return nextUrlParams.term !== prevUrlParams.term ||
+      (nextUrlParams.tab == null && nextUrlParams.index == null);
```

Comment:
This is for the case where the user resubmits the same query. We should redo the search and reset the tab and index. Another way we can do this is to add a `&submit=1` to the url from SearchBar. You can see a sample of this in https://github.com/lyft/amundsenfrontendlibrary/pull/236/files

Reply:
I think I understand the case, but I'm not sure why this logic is here in this method. I see this and I think "Why would a method to update the `searchTerm` be in any way dependent on values for `tab` and `index`?"

Reply:
When the user submits the search bar, the page URL changes to `search?searchTerm=X`. In this case we should re-execute a search all. What happens without this line is that the search doesn't execute, and then `shouldUpdateTab` returns true and sets the `selectedTab` to `null`, since the page URL no longer contains a `selectedTab`.

Perhaps this function is best renamed `shouldSearchAll` instead of `shouldUpdateSearchTerm`.